### PR TITLE
Add assets in pool with native to query_acceptable_payment_assets's return

### PR DESCRIPTION
**Pull request by franciscoaguirre**
`XcmPaymentApi::query_acceptable_payment_assets` is an API that returns the assets that can be used to pay fees on the runtime where it's called.
For relays and most system chains this was configured only as the native asset: ROC and WND.
However, the asset hubs have the asset conversion pallet, which allows fees to be paid in any asset in a pool with the native one.
This PR adds the list of assets in a pool with the native one to the return value of this API for the asset hubs.

### CODE REVIEW

**Review by acatangiu — APPROVED** [review]

File: cumulus/parachains/runtimes/assets/asset-hub-rococo/src/lib.rs [hunk 1]
```diff
@@ -1326,7 +1326,19 @@ impl_runtime_apis! {
...
+			];
+			// We also accept all assets in a pool with the native token.
+			let keys_iter = pallet_asset_conversion::Pools::<Runtime>::iter_keys();
+			for (asset_1, asset_2) in keys_iter {
+				if asset_1 == xcm_config::TokenLocation::get() {
+					acceptable_assets.push(asset_2.clone().into());
+				} else if asset_2 == xcm_config::TokenLocation::get() {
+					acceptable_assets.push(asset_1.clone().into());
+				}
+			}
```
Comment (bkontur) [inline cumulus/parachains/runtimes/assets/asset-hub-rococo/src/lib.rs:1341]: just a very small nit, I would avoid constructing `TokenLocation::get()` on every iteration twice,
also probably `extend` could be more a little bit more efficient then `push` here:

```suggestion
			let native_token = xcm_config::TokenLocation::get();
			// We accept the native token to pay fees.
			let mut acceptable_assets = vec![AssetId(native_token.clone())];
			// We also accept all assets in a pool with the native token.
			acceptable_assets.extend(
				pallet_asset_conversion::Pools::<Runtime>::iter_keys().filter_map(
					|(asset_1, asset_2)| {
						if asset_1 == native_token {
							Some(asset_2.clone().into())
						} else if asset_2 == native_token {
							Some(asset_1.clone().into())
						} else {
							None
						}
					},
				),
			);
```


also it is possible to write it as a one-liner, but not sure, which one is more readable:

```
		fn query_acceptable_payment_assets(xcm_version: xcm::Version) -> Result<Vec<VersionedAssetId>, XcmPaymentApiError> {
			let native_token = xcm_config::TokenLocation::get();

			PolkadotXcm::query_acceptable_payment_assets(
				xcm_version,
				// We accept the native token to pay fees.
				core::iter::once(AssetId(native_token.clone()))
					// We also accept all assets in a pool with the native token.
					.chain(pallet_asset_conversion::Pools::<Runtime>::iter_keys().filter_map(
						|(asset_1, asset_2)| {
							if asset_1 == native_token {
								Some(asset_2.clone().into())
							} else if asset_2 == native_token {
								Some(asset_1.clone().into())
							} else {
								None
							}
						},
					))
					.collect(),
			)
		}
```

File: cumulus/parachains/runtimes/assets/asset-hub-westend/src/lib.rs [hunk 2]
```diff
@@ -1357,7 +1357,17 @@ impl_runtime_apis! {
...
+			let mut acceptable_assets = vec![AssetId(xcm_config::WestendLocation::get())];
+			// We also accept all assets in a pool with the native token.
+			let keys_iter = pallet_asset_conversion::Pools::<Runtime>::iter_keys();
+			for (asset_1, asset_2) in keys_iter {
+				if asset_1 == xcm_config::WestendLocation::get() {
+					acceptable_assets.push(asset_2.clone().into());
+				} else if asset_2 == xcm_config::WestendLocation::get() {
+					acceptable_assets.push(asset_1.clone().into());
+				}
+			}
```
Comment (bkontur) [inline cumulus/parachains/runtimes/assets/asset-hub-westend/src/lib.rs:1370]:
```suggestion
			let native_token = xcm_config::WestendLocation::get();
			// We accept the native token to pay fees.
			let mut acceptable_assets = vec![AssetId(native_token.clone())];
			// We also accept all assets in a pool with the native token.
			acceptable_assets.extend(
				pallet_asset_conversion::Pools::<Runtime>::iter_keys().filter_map(
					|(asset_1, asset_2)| {
						if asset_1 == native_token {
							Some(asset_2.clone().into())
						} else if asset_2 == native_token {
							Some(asset_1.clone().into())
						} else {
							None
						}
					},
				),
			);
```

**Review by bkontur — APPROVED** [review]

**Comment by mrshiposha** [comment]
@franciscoaguirre, do you plan to adjust the implementation of the `query_weight_to_asset_fee` so it can calculate the current price in the selected currency using pools?

**Comment by franciscoaguirre** [comment]
Yes, coming right up!

**Comment by franciscoaguirre** [comment]
Wrote up a draft: https://github.com/paritytech/polkadot-sdk/pull/6080

**Pull request merged**